packages/foam-vscode/src/services/variable-resolver.ts

```diff
@@ -271,6 +203,111 @@ export class Resolver {
     return result;
   }
 }
+export class Resolver {
```

Comment:
Have you split the `FoamVariableResolver` from the `Resolver` to make the VS Code API happy?

I am trying to wrap my mind around the individual responsibilities and how tied they are

Reply:
> to make the VS Code API happy?

Not explicitly. I did it for convenience, since that way I didn't have to change the consumers of `Resolver`. Probably worth a revisit to see if it makes sense to merge the two.

Reply:
Alright, that was easy. Just had to rename `resolve` to `resolveForName`.

Done 👍